src/main/java/seedu/uninurse/model/person/PatientMatchPredicate.java

```diff
@@ -0,0 +1,90 @@
+package seedu.uninurse.model.person;
+
+import java.util.List;
+import java.util.function.Predicate;
+
+import seedu.uninurse.model.condition.ConditionContainsKeywordsPredicate;
+import seedu.uninurse.model.tag.TagContainsKeywordsPredicate;
+import seedu.uninurse.model.task.TaskContainsKeywordsPredicate;
+
+/**
+ * Tests that a {@code Patient} matches with all descriptors.
+ */
+public class PatientMatchPredicate implements Predicate<Patient> {
+    private final PatientContainsKeywordsPredicate patientContainsKeywordsPredicate;
+    private final NameContainsKeywordsPredicate nameContainsKeywordsPredicate;
+    private final PhoneContainsKeywordsPredicate phoneContainsKeywordsPredicate;
+    private final EmailContainsKeywordsPredicate emailContainsKeywordsPredicate;
+    private final AddressContainsKeywordsPredicate addressContainsKeywordsPredicate;
+    private final TagContainsKeywordsPredicate tagContainsKeywordsPredicate;
+    private final ConditionContainsKeywordsPredicate conditionContainsKeywordsPredicate;
+    private final TaskContainsKeywordsPredicate taskContainsKeywordsPredicate;
+
+    /**
+     * Constructs a {@code PatientMatchPredicate}
+     * which tests {@code Patient} with given keywords.
+     *
+     * @param keywords
+     */
+    public PatientMatchPredicate(List<String> keywords) {
+        this(keywords, List.of(""), List.of(""), List.of(""),
+                List.of(""), List.of(""), List.of(""), List.of(""));
+    }
+
+    /**
+     * Constructs a {@code PatientMatchPredicate}
+     * which tests {@code Patient} with all given descriptors.
+     *
+     * @param keywords
+     * @param nameList
+     * @param phoneList
+     * @param emailList
+     * @param addressList
+     * @param tagList
+     * @param conditionList
+     * @param taskList
+     */
+    public PatientMatchPredicate(List<String> keywords, List<String> nameList, List<String> phoneList,
+                                 List<String> emailList, List<String> addressList, List<String> tagList,
+                                 List<String> conditionList, List<String> taskList) {
+        this.patientContainsKeywordsPredicate = new PatientContainsKeywordsPredicate(keywords);
+        this.nameContainsKeywordsPredicate = new NameContainsKeywordsPredicate(nameList);
+        this.phoneContainsKeywordsPredicate = new PhoneContainsKeywordsPredicate(phoneList);
+        this.emailContainsKeywordsPredicate = new EmailContainsKeywordsPredicate(emailList);
+        this.addressContainsKeywordsPredicate = new AddressContainsKeywordsPredicate(addressList);
+        this.tagContainsKeywordsPredicate = new TagContainsKeywordsPredicate(tagList);
+        this.conditionContainsKeywordsPredicate = new ConditionContainsKeywordsPredicate(conditionList);
+        this.taskContainsKeywordsPredicate = new TaskContainsKeywordsPredicate(taskList);
+    }
```

Comment:
I think you can just use the existing `ArgumentMultimap` or maybe even create a new `KeywordMultimap` here.

Reply:
Done